src/main/java/jimmy/mcgymmy/logic/parser/ParserUtil.java

```diff
@@ -31,6 +31,8 @@
 public class ParserUtil {
 
     public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
+    public static final String MESSAGE_INDEX_TOO_LARGE = "Index must be smaller than 2^31.";
```

Comment:
I think our limit for values is 1000, there's a function in `MacroNutrient` that says
```java
private String getMessageConstraint() {
        return this.getMacronutrientType() + " amount can only take in value larger than 0 and less than 1000";
    }
```

Also I think we should make all the error messages use this function somehow, so we have DRY. see #214

Reply:
I think this is for the index, not the macronutrient value?

Reply:
I'm quite sure its for all of those though? either way we have to limit all of them to much lower than MAX_INT

Reply:
> I think this is for the index, not the macronutrient value?

As Joey mentioned correctly, we are limiting macronutrient values to 1000g each. Firstly because exceeding 1kg of any type of nutrient for a single dish does not make much sense so limiting it to 1000g does not limit its utility. Second and more importantly, exceeding 1000g of macronutrient will increase total calorie count by a lot, which in turn will cause the total calorie count to reach integer overflow easily. 

Basically, yeah we should limit protein, carb, fat value to 1000g.

Reply:
ok, I will limit protein, carb, and fat value to 1000. But do we really need to also limit index value to 1000 too?

Reply:
I don't think its for index. It only applies to MacroNutrients